### PR TITLE
planner: use code-gen to generate CloneForPlanCache method for Point/BatchPoint/Limit

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: use code-gen to generate CloneForPlanCache method for some operators

### What changed and how does it work?

planner: use code-gen to generate CloneForPlanCache method for some operators

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/planner/core/plan_clone_generated.go [hunk 1]
```diff
@@ -222,11 +222,83 @@ func (op *PhysicalIndexReader) CloneForPlanCache(newCtx base.PlanContext) (base.
...
+	}
+	cloned.IndexValues = util.CloneDatums(op.IndexValues)
+	cloned.IndexConstants = util.CloneConstants(op.IndexConstants)
+	cloned.IdxCols = util.CloneColumns(op.IdxCols)
+	cloned.IdxColLens = make([]int, len(op.IdxColLens))
+	copy(cloned.IdxColLens, op.IdxColLens)
+	cloned.AccessConditions = util.CloneExpressions(op.AccessConditions)
+	cloned.ctx = newCtx
+	cloned.accessCols = util.CloneColumns(op.accessCols)
+	probeParents, ok := clonePhysicalPlansForPlanCache(newCtx, op.probeParents)
```
Comment (AilinKid) [inline pkg/planner/core/plan_clone_generated.go:254]: we can move this a little bit up for quick fail over

File: pkg/planner/core/plan_clone_generated.go [hunk 2]
```diff
@@ -222,11 +222,83 @@ func (op *PhysicalIndexReader) CloneForPlanCache(newCtx base.PlanContext) (base.
...
+	cloned.IndexValues = util.CloneDatum2D(op.IndexValues)
+	cloned.IndexValueParams = util.CloneConstant2D(op.IndexValueParams)
+	cloned.AccessConditions = util.CloneExpressions(op.AccessConditions)
+	cloned.IdxCols = util.CloneColumns(op.IdxCols)
+	cloned.IdxColLens = make([]int, len(op.IdxColLens))
+	copy(cloned.IdxColLens, op.IdxColLens)
+	cloned.PartitionIdxs = make([]int, len(op.PartitionIdxs))
+	copy(cloned.PartitionIdxs, op.PartitionIdxs)
+	cloned.accessCols = util.CloneColumns(op.accessCols)
+	probeParents, ok := clonePhysicalPlansForPlanCache(newCtx, op.probeParents)
```
Comment (AilinKid) [inline pkg/planner/core/plan_clone_generated.go:280]: same here

File: pkg/planner/core/plan_clone_generated.go (hunk 1 above)
Comment (qw4990) [reply pkg/planner/core/plan_clone_generated.go]: Updated. Thx.

**Review by time-and-fate — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Pull request merged**